File: backend/ExpenseControl.Domain/Entities/Transaction.cs
using ExpenseControl.Domain.Enums;

namespace ExpenseControl.Domain.Entities;

public class Transaction
{
    public int Id { get; private set; }

    public string Description { get; private set; } = string.Empty;

    public decimal Amount { get; private set; }

    public TransactionType Type { get; private set; }

    public int PersonId { get; private set; }

    // Referência para a pessoa responsável pela movimentação.
    // O vínculo garante que toda transação pertença a um cadastro existente.
    public Person Person { get; private set; } = null!;

    private Transaction()
    {
        // Necessário para o Entity Framework Core reconstruir
        // objetos ao consultar os dados persistidos.
    }

    public Transaction(
        string description,
        decimal amount,
        TransactionType type,
        int personId)
    {
        Description = description;
        Amount = amount;
        Type = type;
        PersonId = personId;
    }
}

File: frontend/src/types/transaction.ts
export const TransactionType = {
  Expense: 1,
  Income: 2,
} as const;

export type TransactionType =
  (typeof TransactionType)[keyof typeof TransactionType];

export interface Transaction {
  id: number;
  description: string;
  amount: number;
  type: TransactionType;
  personId: number;
}

export interface CreateTransaction {
  description: string;
  amount: number;
  type: TransactionType;
  personId: number;
}